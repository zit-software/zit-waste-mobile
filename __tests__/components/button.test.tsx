import renderer from 'react-test-renderer';

import Button, { ButtonProps } from '../../components/button';

describe('<Button />', () => {
	let defaultProps: ButtonProps;

	beforeAll(() => {
		defaultProps = {
			title: 'Button',
		};
	});

	it('should be render', () => {
		const tree = renderer
			.create(<Button {...defaultProps} />)
			.toJSON() as renderer.ReactTestRendererJSON;

		expect(tree).toMatchSnapshot();
	});

	it('should be render with icon', () => {
		const tree = renderer
			.create(<Button {...defaultProps} icon='icon' />)
			.toJSON() as renderer.ReactTestRendererJSON;

		expect(tree).toMatchSnapshot();
	});

	it('should pressable', () => {
		let pressed = false;

		function handlePress() {
			pressed = true;
		}

		const tree = renderer.create(
			<Button {...defaultProps} onPress={handlePress} />,
		);

		tree.root.props.onPress();

		expect(pressed).toBeTruthy();
	});
});
