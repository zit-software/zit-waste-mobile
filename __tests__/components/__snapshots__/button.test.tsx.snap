// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be render 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00000020",
      "borderRadius": 15,
      "display": "flex",
      "flexDirection": "row",
      "gap": 5,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat_700Bold",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;

exports[`<Button /> should be render with icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00000020",
      "borderRadius": 15,
      "display": "flex",
      "flexDirection": "row",
      "gap": 5,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 25,
      "paddingVertical": 15,
    }
  }
>
  icon
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat_700Bold",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;
